### PR TITLE
LPD-43040 Create fetchDDMTemplateByExternalReferenceCode method including ancestors, since we have the need to get template from global, current site or related asset libraries

### PR DESCRIPTION
**Pull request by Mikellorza**
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-43040

# How am I fixing it?

Add fetchDDMTemplateByExternalReferenceCode including ancestors, since we have the need to get template from global, current site or related asset libraries

# How can you verify that it works?

Run the included  test.

### CODE REVIEW

**Comment by marcelabc** [comment]
Just started reviewing :)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/DDMTemplateLocalServiceTest.java [hunk 1]
```diff
@@ -215,6 +217,33 @@ public void testDeleteTemplateByExternalReferenceCode() throws Exception {
 			_ddmTemplateLocalService.fetchTemplate(template.getTemplateId()));
 	}
 
+	@Test
+	public void testFetchDDMTemplateByExternalReferenceCodeIncludingAncestors()
```
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/DDMTemplateLocalServiceTest.java:221]: Could you rename it to: testFetchDDMTemplateByExternalReferenceCodeIncluding**AncestorTemplates**

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/DDMTemplateLocalServiceTest.java [hunk 2]
```diff
@@ -215,6 +217,33 @@ public void testDeleteTemplateByExternalReferenceCode() throws Exception {
...
+		throws Exception {
+
+		Company company = _companyLocalService.getCompany(group.getCompanyId());
+
+		DDMTemplate template = _ddmTemplateLocalService.addTemplate(
```
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/DDMTemplateLocalServiceTest.java:226]: Could you please delete the `template` at the end of your test?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by marcelabc** [comment]
Hey @Mikellorza 
I'm not sure if you saw that I finished the review! :sweat_smile: 
Are you planning to apply the suggestions in the review? Let me know if you need help!

**Comment by Mikellorza** [comment]
Hi @marcelabc ,

I was on PTO last week, so , today I returned and I am going to apply your suggestions.

Regards!

**Comment by Mikellorza** [comment]
Hi again @marcelabc ,

I have made the changes you proposed and I 'm fine with them. So, it the PR looks great to you we can move forward.

Thanks for your work!

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by Mikellorza** [comment]
ci:report:87563206

**Comment by Mikellorza** [comment]
ci:report:87563206

**Pull request closed without merging**